include/vcpkg/base/util.h

```diff
@@ -130,12 +130,57 @@ namespace vcpkg::Util
         return ret;
     }
 
+    template<class Container, class T>
+    void erase_remove(Container& cont, const T& el)
+    {
+        cont.erase(std::remove(cont.begin(), cont.end(), el), cont.end());
+    }
     template<class Container, class Pred>
     void erase_remove_if(Container& cont, Pred pred)
     {
         cont.erase(std::remove_if(cont.begin(), cont.end(), pred), cont.end());
     }
 
+    // 0th is the first occurence
+    // so find_nth({1, 2, 1, 3, 1, 4}, 1, 2)
+    // returns the 1 before the 4
+    template <class R, class V>
+    auto find_nth(R&& r, const V& v, size_t n)
+    {
+        using std::begin;
+        using std::end;
+
+        auto last = end(r);
+        auto current = std::find(begin(r), last, v);
+
+        for (size_t i = 0; i < n && current != last; ++i)
+        {
+            current = std::find(current + 1, last, v);
+        }
+
+        return current;
+    }
+
+    // 0th is the last occurence
+    // so find_nth({1, 2, 1, 3, 1, 4}, 1, 2)
+    // returns the 1 before the 2
+    template <class R, class V>
+    auto find_nth_from_last(R&& r, const V& v, size_t n)
+    {
+        using std::rbegin;
+        using std::rend;
+
+        auto last = rend(r);
+        auto current = std::find(rbegin(r), last, v);
+
+        for (size_t i = 0; i < n && current != last; ++i)
+        {
+            current = std::find(current + 1, last, v);
+        }
+
+        return current.base() - 1;
```

Comment:
I thought you didn't want a reverse_iterator solution? How about something like this:

```
template<class BidIt, class NeedleTy>
BidIt find_nth_from_last(BidIt first, const BidIt last, const NeedleTy& needle, size_t n) {
    if (first != last) {
        BidIt next = last;
        do {
            --next;
            if (needle == *next) {
                if (n == 0) {
                    return next;
                }

                --n;
            }
        } while (first != next);
    }

    return last;
}

template <class R, class NeedleTy>
auto find_nth_from_last(R& haystack, const NeedleTy& needle, size_t n) {
    using std::begin;
    using std::end;
    return find_nth_from_last(begin(haystack), end(haystack), needle, n);
}
```

https://gcc.godbolt.org/z/Yc9PT1f5c

Reply:
It was more about less thinky when using the thing, not necessarily removing all uses of reverse iterators in the implementation code.